Kernel/Net/NetworkTask.cpp

```diff
@@ -612,6 +613,18 @@ void handle_tcp(IPv4Packet const& ipv4_packet, UnixDateTime const& packet_timest
             // FIXME: Verify that this transition is legitimate.
             socket->set_state(TCPSocket::State::Closed);
             return;
+        case TCPFlags::ACK:
+            socket->set_duplicate_acks(0);
+            if (payload_size) {
+                if (socket->did_receive(ipv4_packet.source(), tcp_packet.source_port(), { &ipv4_packet, sizeof(IPv4Packet) + ipv4_packet.payload_size() }, packet_timestamp)) {
+                    socket->set_ack_number(tcp_packet.sequence_number() + payload_size);
```

Comment:
Hm. Why should we do seq number + payload_size here, and not the +1 from other states? Looking at other states it's not clear to me when we add 1 to the ACK number and when we don't? If we do add 1, do we still need duplicate acks enabled?

Reply:
Yeah, it should be + 1 actually. 

>If we do add 1, do we still need duplicate acks enabled?

Yep, I believe that's the case. I'll add the changes.